### PR TITLE
GAWB-3293 return Cromwell Subworkflow Metadata

### PR DESCRIPTION
**Pull request by jmthibault79**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala [hunk 1]
```diff
@@ -59,7 +59,7 @@ class HttpExecutionServiceDAO(executionServiceURL: String, override val workbenc
   }
 
   override def callLevelMetadata(id: String, userInfo: UserInfo): Future[JsObject] = {
-    val url = executionServiceURL + s"/api/workflows/v1/${id}/metadata"
+    val url = executionServiceURL + s"/api/workflows/v1/${id}/metadata?expandSubWorkflows=true"
```
Comment (jmthibault79) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala:62]: Here's the only production code change.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        |
+        |  runtime {
+        |	docker: docker_image
+        |  }
+        |
+        |}
+      """.stripMargin)
+
+  def subWorkflowParentMethod(child: Method): Method = {
+    val childUrl = s"http://orch-app:8080/ga4gh/v1/tools/${child.methodNamespace}:${child.methodName}/versions/1/plain-WDL/descriptor"
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala]: woah. cool that this works

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 3]
```diff
@@ -81,5 +87,88 @@ class RawlsApiSpec extends FreeSpec with Matchers with CleanUp with BillingFixtu
         }
       }
     }
+
+    object SingleParticipant {
+      val participantEntity = "entity:participant_id\nparticipant1"
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: and the most minimal TSV award goes to... :trophy:

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        |	File inputBam
+        |	File inputBamIndex
+        |
+        |	String gvcf_name
+        |	String intervals
+        |
+        |	String docker_image
+        |
+        |	command {
+        |		gatk HaplotypeCaller \
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala:75]: what's the data we're running this on for the test? does it run in tiny-time? i wonder if hand-rolling an echo-strings wdl would be faster

**Review by helgridly — COMMENTED** [review]
I am thumb-sideways on running a workflow that does a thing, simply because it might take a while. Or at least longer than `echo okay done`.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala (hunk 3 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala]: stole this from somewhere

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala (hunk 2 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala]: this took many iterations.

...but now it occurs to me that this is not likely to work on Alpha.  Hrm.

Comment (aednichols) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala]: Use the mock server? That’s what I did for Dockstore (search around and you’ll find it)

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala (hunk 4 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala:75]: It's pretty small.  Even so, it aborts as soon as it detects that there's subworkflow metadata.

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala (hunk 2 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala]: I'd rather not use mocks in an Integration Test.

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala]: Trying a new version that uses this for FIABs but the proper Orch API URL for real envs

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/TestMethods.scala]: Success!  Ran on Alpha and failed in the correct way (import succeeded, and there were no subWorkflows in metadata.  Need this to merge for that to pass.)

**Comment by jmthibault79** [comment]
https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/3587/

One jenkins-auto-test failure: Leonardo "Creating 1 worker cluster should return server error" - this test is a known issue that we are working on elsewhere.  It has no connection to Rawls or Cromwell.

**Review by helgridly — APPROVED** [review]

**Review by aweng98 — APPROVED** [review]
Tests ran perfectly on dev. But running test on alpha on this branch encountered a known error `Sam call to HttpMethod(DELETE) /api/resource/billing-project/gpalloc-dev-master-i4o9kec failed"`. I think it's unrelated to this test because I've seen error happens in other tests. 👍 on new working test.
Maybe you can find a fix.

**Comment by aweng98** [comment]
I used cmd to run rawls tests on alpha
`./run-tests.sh alpha rawls jt_3293 alpha` from `firecloud-automated-testing` repo folder.

**Pull request merged**